### PR TITLE
List umbrella child top level deps in hex.outdated

### LINKED ISSUE #296: mix hex.outdated in umbrella root doesn't list top-level packages
elixir/erlang:

```
Erlang/OTP 19 [erts-8.1] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.4.0-dev (470b2ee)
```

expected:

```
~ umbrella_example $ mix hex.outdated
plug         1.2.2       1.2.2       ~> 1.2
poison       3.0.0       3.0.0       ~> 3.0
postgrex     0.12.1      0.12.1      >= 0.0.0
```

actual:

```
~ umbrella_example $ mix hex.outdated
No hex dependencies
```

It's worth noting that `mix hex.outdated --all` does return all packages though.

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #296

cc @obahareth

### CODE REVIEW

File: lib/mix/tasks/hex.outdated.ex [hunk 1]
```diff
@@ -134,6 +136,20 @@ defmodule Mix.Tasks.Hex.Outdated do
...
 
+  defp top_level_deps(deps) do
+    Enum.filter(deps, & &1.top_level) ++ umbrella_top_level_deps(deps)
+  end
+
+  def umbrella_top_level_deps(deps) do
+    if Mix.Project.umbrella?() do
+      apps_paths = Enum.map(Mix.Project.apps_paths(), fn {_, path} -> Path.join(File.cwd!(), path) end)
+
+      Enum.filter(deps, &Path.dirname(&1.from) in apps_paths)
```
Comment (wojtekmach) [inline lib/mix/tasks/hex.outdated.ex]: this is super ugly, need to find a better way to mark child's top-level deps. What this is doing is comparing e.g.:

```
%{from: "/Users/wojtek/src/hexpm/hex/tmp/umbrella/apps/bacon/mix.exs", ...},
%{from: "/Users/wojtek/src/hexpm/hex/tmp/umbrella/deps/bar/mix.exs", ...},
...
```

against /path/to/apps__

Comment (obahareth) [reply lib/mix/tasks/hex.outdated.ex]: What other ways are there to identify them other than the path?

Comment (wojtekmach) [reply lib/mix/tasks/hex.outdated.ex]: that's what I'm trying to figure out :)

File: lib/mix/tasks/hex.outdated.ex [hunk 2]
```diff
@@ -134,6 +136,20 @@ defmodule Mix.Tasks.Hex.Outdated do
...
 
+  defp top_level_deps(deps) do
+    Enum.filter(deps, & &1.top_level) ++ umbrella_top_level_deps(deps)
+  end
+
+  def umbrella_top_level_deps(deps) do
+    if Mix.Project.umbrella?() do
+      apps_paths = Path.expand(Mix.Project.config[:apps_path], File.cwd!())
+
+      Enum.filter(deps, &String.contains?(Path.dirname(Path.dirname(&1.from)), apps_paths))
```
Comment (ericmj) [inline lib/mix/tasks/hex.outdated.ex]: Doesn't this assume that the umbrella children are two directories below the current project?

Comment (ericmj) [reply lib/mix/tasks/hex.outdated.ex]: No never mind, I see how it works now.

**Pull request merged**